### PR TITLE
feat: circuit breaker for smart accounts

### PR DESCRIPTION
**Pull request by PaddyMc**
Closes: #XXX

## What is the purpose of the change

Adding circuit breaker params in the store, this enables governance to turn off smart accounts

## How to test

- `cd ./x/authentictors`
- `go test ./...`
- `go test ./... -run TestAuthenticatorSuite/TestCircuitBreakerStory`

### CODE REVIEW

File: app/ante.go [hunk 1]
```diff
@@ -67,7 +67,17 @@ func NewAnteHandler(
...
-		authante.NewAuthenticatorDecorator(authenticatorKeeper, accountKeeper, signModeHandler),
+		authante.NewAuthenticatorDecorator(authenticatorKeeper, accountKeeper, signModeHandler,
+			// Both the signature verification and gas consumption handlers functionality
+			// is enbedded in the authenticator decorator, we add the both here to enable a
+			// circuit breaker pattern.
+			sdk.ChainAnteDecorators(
+				ante.NewSigGasConsumeDecorator(accountKeeper, sigGasConsumer),
+				ante.NewSigVerificationDecorator(accountKeeper, signModeHandler),
+				ante.NewIncrementSequenceDecorator(accountKeeper),
+				ibcante.NewRedundantRelayDecorator(channelKeeper),
```
Comment (nicolaslara) [inline app/ante.go:78]: what is this? RedundantRelayDecorator? I don't think that was there before

File: app/ante.go [hunk 2]
```diff
@@ -67,7 +67,17 @@ func NewAnteHandler(
 		authante.NewSetPubKeyDecorator(accountKeeper), // SetPubKeyDecorator must be called before all signature verification decorators
 		ante.NewValidateSigCountDecorator(accountKeeper),
 		// Our authenticator decorator
-		authante.NewAuthenticatorDecorator(authenticatorKeeper, accountKeeper, signModeHandler),
+		authante.NewAuthenticatorDecorator(authenticatorKeeper, accountKeeper, signModeHandler,
+			// Both the signature verification and gas consumption handlers functionality
+			// is enbedded in the authenticator decorator, we add the both here to enable a
+			// circuit breaker pattern.
+			sdk.ChainAnteDecorators(
```
Comment (nicolaslara) [inline app/ante.go:74]: maybe we want to give this a name to make it easier to read?

```suggestion
			classicSignatureVerificationDecorator,
```

File: app/ante.go (hunk 1 above)
Comment (PaddyMc) [reply app/ante.go:78]: It's in main atm!  

https://github.com/osmosis-labs/osmosis/blob/main/app/ante.go#L68

File: app/ante.go (hunk 2 above)
Comment (PaddyMc) [reply app/ante.go]: Good input, updated in latest commit

**Pull request merged**